### ISSUE TITLE
Is it possible to have (marginalia?) annotations on async consult commands to display consult group info?

### ISSUE THREAD

**Issue by Dima-369** [minad/consult#455]
This is how it looks (more or less by default, I changed the face in vertico) in `consult-ripgrep`:

![image](https://user-images.githubusercontent.com/15002298/139237245-d5a90f01-cfcd-4848-829f-587c44494851.png)

What I wondered is if one can get a similar file name display like his one from Jetbrains products. Here the file names are annotated to the right where marginalia would usually display its annotations.

![image](https://user-images.githubusercontent.com/15002298/139237332-ddc857ca-d807-4fe8-9b23-248035d4a7f6.png)

I know that this is an aesthetic thing but what I 'dislike' in the current `consult-ripgrep` display is how it jumps over the headings which gets a bit annoying when there are many file names and I often tend to lose the highlighted line when scrolling through.

Basically what I am interested in is moving the consult groups (the file names) to the annotation system. It's just an idea of mine because I personally like how Jetbrains has implemented it.
